plugin/src/main/java/net/aufdemrand/denizen/events/player/PlayerMendsItemScriptEvent.java

```diff
@@ -0,0 +1,143 @@
+    // <context.xp_orb> returns the XP orb that triggered the event.
+    //
+    // @Determine
+    // Element(Number) to set the amount of durability the item recovers.
+    //
+    // -->
+
+    public PlayerMendsItemScriptEvent() {
+        instance = this;
+    }
+
+    public static PlayerMendsItemScriptEvent instance;
+    public dItem item;
+    public dEntity experienceOrb;
+    public Element repairAmount;
+    public PlayerItemMendEvent event;
+    public dLocation location;
+
+    @Override
+    public boolean couldMatch(ScriptContainer scriptContainer, String s) {
+        return CoreUtilities.toLowerCase(s).startsWith("player mends");
+    }
+
+    @Override
+    public boolean matches(ScriptContainer scriptContainer, String s) {
+        String lower = CoreUtilities.toLowerCase(s);
+
+        String iItem = CoreUtilities.getXthArg(2, lower);
+        if (!tryItem(item, iItem)) {
+            return false;
+        }
+        if (!runInCheck(scriptContainer, s, lower, location)) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return "PlayerMendsItem";
+    }
+
+    @Override
+    public void init() {
+        Bukkit.getServer().getPluginManager().registerEvents(this, DenizenAPI.getCurrentInstance());
+    }
+
+    @Override
+    public void destroy() {
+        PlayerItemMendEvent.getHandlerList().unregister(this);
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptContainer container, String determination) {
+        if (aH.matchesInteger(determination)) {
+            repairAmount = new Element(determination);
+            return true;
+        }
+        return super.applyDetermination(container, determination);
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(dPlayer.mirrorBukkitPlayer(event.getPlayer()), null);
+    }
+
+    @Override
+    public dObject getContext(String name) {
+        if (name.equals("item")) {
+            return item;
+        }
+        else if (name.equals("repair_amount")) {
+            return repairAmount;
+        }
+        else if (name.equals("xp_orb")) {
+            return experienceOrb;
+        }
+        return super.getContext(name);
+    }
+
+    @EventHandler
+    public void onPlayerItemMend(PlayerItemMendEvent event) {
+        if (dEntity.isNPC(event.getPlayer())) {
+            return;
+        }
+        item = new dItem(event.getItem());
+        experienceOrb = new dEntity(event.getExperienceOrb());
+        location = new dLocation(event.getPlayer().getLocation());
+        repairAmount = new Element(event.getRepairAmount());
+        cancelled = event.isCancelled();
+        this.event = event;
+        fire();
+        event.setCancelled(cancelled);
+        event.setRepairAmount(repairAmount.asInt());
+        final Player p = event.getPlayer();
+        if (cancelled) {

```

Comment:
pretty sure this whole bit isn't needed

Reply:
I'm pretty sure lines 130 and 131 are required to apply the determinations, though everything after line 132 might be unneeeded. I'll test that some time this week.

Reply:
Confirmed that lines 132 to 140 are unnecessary (credits to @mergu). Will remove those lines at some point tomorrow night.